core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -603,6 +603,15 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
    */
   protected def doKillExecutors(executorIds: Seq[String]): Future[Boolean] =
     Future.successful(false)
+
+  /**
+   * Request that the cluster manager kill all executors on a given host.
+   * @return whether the kill request is acknowledged
+   */
+  final override def killExecutorsOnHost(host: String): Seq[String] = {
+    logInfo(s"Requesting to kill any and all executors on host ${host}")
+    killExecutors(scheduler.getExecutorsAliveOnHost(host).get.toSeq, replace = true, force = true)
```

Comment:
it might be possible that `scheduler.getExecutorsAliveOnHost` returns None -- the executor happens to get removed between when it gets blacklisted and when you end up here.

We might prevent that by the locks we have & the way msgs are processed, but it would be better to be safe on this anyhow:

```scala
scheduler.getExecutorsAliveOnHost(host).foreach(killExecutors(_.toSeq, ...))
```

I am also now wondering if we should kill all the executors immediately, or if we should push a request into one of the internal queues in the driver.  Killing itself is async, so maybe it fine like this. I'm not really sure yet, something to keep thinking about.